### DISCUSSION TITLE
v2.1.0 - Goals & Custom Properties broken

### DISCUSSION THREAD

**Discussion by Olakeen** [plausible/analytics#4169] [Self-Hosted Support] [answered]
I understand that on this new version, Custom Properties are no longer added automatically, but the button doesn't work.

![1](https://github.com/plausible/analytics/assets/26546237/9eff6034-b60f-45ee-bb59-716630b70ded)

Also, the Goals menu also seems inoperative
![2](https://github.com/plausible/analytics/assets/26546237/ea32f183-6d35-4efe-a0c5-1bce55e042f9)

I tried it on another browser and in private browsing, the result was the same.

**Comment by ruslandoga** [comment]
👋 @Olakeen 

Thank you for the report and screen recordings. I'll be looking into this today!

**Reply by ruslandoga** [reply]
I'm running v2.1.0 on https://who.copycat.fun and I was able to add a custom property. Please share your browser console logs and `docker compose logs` from when you are clicking the "Add property" button.

**Reply by ruslandoga** [reply]
"Goals" menu seems to work fine too.

**Reply by Olakeen** [reply]
![2024-06-02 114948](https://github.com/plausible/analytics/assets/26546237/09940e4c-f577-4bc0-8a16-34887eba9fe1)

There is no error in the web browser when clicking on Add Property

**Reply by Olakeen** [reply]
I have this message in the browser but not related to a click action
![Sans titre](https://github.com/plausible/analytics/assets/26546237/2ab15ebb-1215-4218-b473-532b119beee6)

**Reply by ruslandoga** [reply, marked as answer]
This form uses WebSockets to interactively fetch data, it doesn't work without it.
Please make sure your reverse-proxy allows WebSocket connections.

NGINX example: https://github.com/plausible/community-edition/blob/c5f3386338057878648fd30af9c0462d6de78425/reverse-proxy/nginx/plausible#L12-L17

**Reply by Olakeen** [reply]
That was it! Thank you very much, it works now.
